core/deployment/src/main/java/io/quarkus/runner/bootstrap/AugmentActionImpl.java

```diff
@@ -348,10 +348,13 @@ private BuildResult runAugment(boolean firstRun, Set<String> changedResources,
             Class<? extends BuildItem>... finalOutputs) {
         ClassLoader old = Thread.currentThread().getContextClassLoader();
         try {
-            Thread.currentThread().setContextClassLoader(curatedApplication.getAugmentClassLoader());
-            ProfileManager.setLaunchMode(launchMode);
-
             QuarkusClassLoader classLoader = curatedApplication.getAugmentClassLoader();
+            Thread.currentThread().setContextClassLoader(classLoader);
+            ProfileManager.setLaunchMode(launchMode);
+            ProfileManager.setRuntimeDefaultProfile(
+                    Optional.ofNullable(quarkusBootstrap.getBuildSystemProperties())
+                            .map(properties -> properties.getProperty(ProfileManager.QUARKUS_PROFILE_PROP))
+                            .orElse(null));
```

Comment:
This means that `ProfileManager.QUARKUS_PROFILE_PROP` present in `quarkusBootstrap.getBuildSystemProperties()` will have lower priority than `System.getProperties()` and the corresponding env var, right?

Reply:
@aloubyansky Exactly, I wanted first to have a higher precedence using a new static field but it brakes a test as I explained in my previous comment, so to avoid breaking anything, I use the runtime default profile instead.

Reply:
You prefer a higher precedence or it is good enough like this?